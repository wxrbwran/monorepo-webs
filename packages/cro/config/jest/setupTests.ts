// jest-dom adds custom jest matchers for asserting on DOM nodes.
// allows you to do things like:
// expect(element).toHaveTextContent(/react/i)
// learn more: https://github.com/testing-library/jest-dom
import '@testing-library/jest-dom/extend-expect';

jest.mock('@/services/auth', () => {
  const API = {
    get: jest.fn(() =>
      Promise.resolve({
        data: {},
        status: 'success',
      }),
    ),
    post: jest.fn(() =>
      Promise.resolve({
        data: {},
        status: 'success',
      }),
    ),
  };
  return {
    ajax: API,
    auth: API,
  };
});

jest.mock('@/services/http', () => {
  const API = {
    get: jest.fn(() =>
      Promise.resolve({
        data: {},
        status: 'success',
      }),
    ),
    post: jest.fn(() =>
      Promise.resolve({
        data: {},
        status: 'success',
      }),
    ),
    setAuthorizationToken: jest.fn(),
  };
  const http = API;
  return http;
});
