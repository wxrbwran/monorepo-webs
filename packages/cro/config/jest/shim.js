import 'core-js/stable';
import 'regenerator-runtime/runtime';

/* eslint-disable import/first */
global.requestAnimationFrame =
  global.requestAnimationFrame ||
  function requestAnimationFrame(callback) {
    setTimeout(callback, 0);
  };

global.matchMedia = window.matchMedia = () => ({
  matches: false,
  addListener: () => { },
  removeListener: () => { },
});
