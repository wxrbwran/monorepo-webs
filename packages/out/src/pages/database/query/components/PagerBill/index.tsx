import React, { useState } from 'react';
import { DownOutlined } from '@ant-design/icons';
import BillType from '../BillType'

import styles from '../../index.scss';

const pargerInfo = [
	{
		key: 'LABORATORY',
		value: '化验单'
	}, {
		key: 'CHECK',
		value: '检查单'
	}
]

const laboratoryType = [
	{
		key: 'SHQX',
		value: '生化全项',
	}, {
		key: 'XCG',
		value: '血常规',
	}, {
		key: 'BCG',
		value: '便常规',
	}
]

const checkType = [
	{
		key: 'XZCS',
		value: '超声',
	}, {
		key: 'XDT',
		value: '心电图',
	}
]

function PagerBill() {

	const [tabKey, setTabKey] = useState('');

	const changeTab = (key: string) => {
		if(key === tabKey) {
			setTabKey('');
		} else {
			setTabKey(key);
		}
	}

	return (
		<div className={styles.main}>
			<div className={styles.lable}>纸质单据</div>
			<div className={styles.group_value} style={tabKey ? { paddingBottom: 180 } : { paddingBottom: 22 }}>
				{
					pargerInfo.map((item) => (
						<div className={styles.item_box} key={item.key}>
							<div
								onClick={() => changeTab(item.key)}
								className={item.key === tabKey ? `${styles.title} ${styles.active}` : styles.title}
							>
								{item.value} <DownOutlined />
							</div>
							<div
								style={item.key === tabKey ? { display: 'block' } : { display: 'none' }}
								className={styles.value}
							>
								{
									item.key === 'LABORATORY' && (
										<BillType
											laboratoryType={laboratoryType}
											toggleTab={()=>setTabKey('')}
										/>
									)
								}
								{
									item.key === 'CHECK' && (
										<BillType
											laboratoryType={checkType}
											toggleTab={()=>setTabKey('')}
										/>
									)
								}
							</div>
						</div>
					))
				}
			</div>
		</div>
	)
}

export default PagerBill;
