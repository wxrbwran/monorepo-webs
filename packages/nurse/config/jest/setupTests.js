// import Enzyme from 'enzyme';
// import Adapter from 'enzyme-adapter-react-16';
import 'jest';
// import axios from 'axios';
// import '@testing-library/jest-dom/extend-expect';

// Enzyme.configure({ adapter: new Adapter() });

jest.mock('@/services/auth', () => {
  const API = {
    get: jest.fn(() =>
      Promise.resolve({
        data: {},
        status: 'success',
      }),
    ),
    post: jest.fn(() =>
      Promise.resolve({
        data: {},
        status: 'success',
      }),
    ),
  };
  return {
    ajax: API,
    auth: API,
  };
});

jest.mock('@/services/http', () => {
  const API = {
    get: jest.fn(() =>
      Promise.resolve({
        data: {},
        status: 'success',
      }),
    ),
    post: jest.fn(() =>
      Promise.resolve({
        data: {},
        status: 'success',
      }),
    ),
    setAuthorizationToken: jest.fn(),
  };
  const http = API;
  return http;
});

// jest.mock('@/services/api', () => {
//     const API = {
//         get: jest.fn(() => Promise.resolve({
//             data: {}, status: 'success',
//         })),
//         post: jest.fn(() => Promise.resolve({
//             data: {}, status: 'success',
//         })),
//         setAuthorizationToken: jest.fn(),
//     };
//     const http = API;
//     return http;
// })
