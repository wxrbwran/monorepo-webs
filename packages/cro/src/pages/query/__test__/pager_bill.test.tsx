import 'jest';
import React from 'react';
import PagerBill from '../pager_bill';
import { render, RenderResult, fireEvent} from '@testing-library/react';
import { Provider } from 'react-redux';
import configureMockStore from 'redux-mock-store';

const mockStore = configureMockStore();
const store = mockStore({
  query: {
    images: [],
    other: [],
    base: {}
  },
});

describe('查询页面：纸质单据', () => {
  let wrapper: RenderResult;
  beforeEach(() => {
    wrapper = render(<Provider store={store}><PagerBill /></Provider>)
  })
  it('匹配快照', () => {
    expect(wrapper).toMatchSnapshot();
  })
  it('点击箭头，输入框显示或隐藏', () => {
    const contInputWrap = wrapper.getByTestId('isShow0');
    expect(contInputWrap).not.toBeVisible();
    const changeTab = wrapper.getByTestId('changeTab0');
    fireEvent.click(changeTab);
    expect(contInputWrap).toBeVisible();
    fireEvent.click(changeTab);
    expect(contInputWrap).not.toBeVisible();
  })
})