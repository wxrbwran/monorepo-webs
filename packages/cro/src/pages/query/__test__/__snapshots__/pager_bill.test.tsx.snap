// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`查询页面：纸质单据 匹配快照 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <div>
          纸质单据
        </div>
        <div
          style="padding-bottom: 22px;"
        >
          <div>
            <div
              data-testid="changeTab0"
            >
              化验单
               
              <span
                aria-label="down"
                class="anticon anticon-down"
                role="img"
              >
                <svg
                  aria-hidden="true"
                  class=""
                  data-icon="down"
                  fill="currentColor"
                  focusable="false"
                  height="1em"
                  viewBox="64 64 896 896"
                  width="1em"
                >
                  <path
                    d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
                  />
                </svg>
              </span>
            </div>
            <div
              data-testid="isShow0"
              style="display: none;"
            >
              <div>
                <div>
                  <label
                    class="ant-checkbox-wrapper"
                  >
                    <span
                      class="ant-checkbox"
                    >
                      <input
                        class="ant-checkbox-input"
                        data-testid="checkbox0"
                        type="checkbox"
                        value=""
                      />
                      <span
                        class="ant-checkbox-inner"
                      />
                    </span>
                    <span>
                      生化全项
                    </span>
                  </label>
                </div>
                <div>
                  <label
                    class="ant-checkbox-wrapper"
                  >
                    <span
                      class="ant-checkbox"
                    >
                      <input
                        class="ant-checkbox-input"
                        data-testid="checkbox1"
                        type="checkbox"
                        value=""
                      />
                      <span
                        class="ant-checkbox-inner"
                      />
                    </span>
                    <span>
                      血常规
                    </span>
                  </label>
                </div>
                <div>
                  <label
                    class="ant-checkbox-wrapper"
                  >
                    <span
                      class="ant-checkbox"
                    >
                      <input
                        class="ant-checkbox-input"
                        data-testid="checkbox2"
                        type="checkbox"
                        value=""
                      />
                      <span
                        class="ant-checkbox-inner"
                      />
                    </span>
                    <span>
                      便常规
                    </span>
                  </label>
                </div>
                <div>
                  <button
                    class="ant-btn"
                    type="button"
                  >
                    <span>
                      取 消
                    </span>
                  </button>
                  <button
                    class="ant-btn ant-btn-primary"
                    type="button"
                  >
                    <span>
                       确定 
                    </span>
                  </button>
                </div>
              </div>
            </div>
          </div>
          <div>
            <div
              data-testid="changeTab1"
            >
              检查单
               
              <span
                aria-label="down"
                class="anticon anticon-down"
                role="img"
              >
                <svg
                  aria-hidden="true"
                  class=""
                  data-icon="down"
                  fill="currentColor"
                  focusable="false"
                  height="1em"
                  viewBox="64 64 896 896"
                  width="1em"
                >
                  <path
                    d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
                  />
                </svg>
              </span>
            </div>
            <div
              data-testid="isShow1"
              style="display: none;"
            >
              <div>
                <div>
                  <label
                    class="ant-checkbox-wrapper"
                  >
                    <span
                      class="ant-checkbox"
                    >
                      <input
                        class="ant-checkbox-input"
                        data-testid="checkbox0"
                        type="checkbox"
                        value=""
                      />
                      <span
                        class="ant-checkbox-inner"
                      />
                    </span>
                    <span>
                      超声
                    </span>
                  </label>
                </div>
                <div>
                  <label
                    class="ant-checkbox-wrapper"
                  >
                    <span
                      class="ant-checkbox"
                    >
                      <input
                        class="ant-checkbox-input"
                        data-testid="checkbox1"
                        type="checkbox"
                        value=""
                      />
                      <span
                        class="ant-checkbox-inner"
                      />
                    </span>
                    <span>
                      心电图
                    </span>
                  </label>
                </div>
                <div>
                  <button
                    class="ant-btn"
                    type="button"
                  >
                    <span>
                      取 消
                    </span>
                  </button>
                  <button
                    class="ant-btn ant-btn-primary"
                    type="button"
                  >
                    <span>
                       确定 
                    </span>
                  </button>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <div>
        纸质单据
      </div>
      <div
        style="padding-bottom: 22px;"
      >
        <div>
          <div
            data-testid="changeTab0"
          >
            化验单
             
            <span
              aria-label="down"
              class="anticon anticon-down"
              role="img"
            >
              <svg
                aria-hidden="true"
                class=""
                data-icon="down"
                fill="currentColor"
                focusable="false"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
                />
              </svg>
            </span>
          </div>
          <div
            data-testid="isShow0"
            style="display: none;"
          >
            <div>
              <div>
                <label
                  class="ant-checkbox-wrapper"
                >
                  <span
                    class="ant-checkbox"
                  >
                    <input
                      class="ant-checkbox-input"
                      data-testid="checkbox0"
                      type="checkbox"
                      value=""
                    />
                    <span
                      class="ant-checkbox-inner"
                    />
                  </span>
                  <span>
                    生化全项
                  </span>
                </label>
              </div>
              <div>
                <label
                  class="ant-checkbox-wrapper"
                >
                  <span
                    class="ant-checkbox"
                  >
                    <input
                      class="ant-checkbox-input"
                      data-testid="checkbox1"
                      type="checkbox"
                      value=""
                    />
                    <span
                      class="ant-checkbox-inner"
                    />
                  </span>
                  <span>
                    血常规
                  </span>
                </label>
              </div>
              <div>
                <label
                  class="ant-checkbox-wrapper"
                >
                  <span
                    class="ant-checkbox"
                  >
                    <input
                      class="ant-checkbox-input"
                      data-testid="checkbox2"
                      type="checkbox"
                      value=""
                    />
                    <span
                      class="ant-checkbox-inner"
                    />
                  </span>
                  <span>
                    便常规
                  </span>
                </label>
              </div>
              <div>
                <button
                  class="ant-btn"
                  type="button"
                >
                  <span>
                    取 消
                  </span>
                </button>
                <button
                  class="ant-btn ant-btn-primary"
                  type="button"
                >
                  <span>
                     确定 
                  </span>
                </button>
              </div>
            </div>
          </div>
        </div>
        <div>
          <div
            data-testid="changeTab1"
          >
            检查单
             
            <span
              aria-label="down"
              class="anticon anticon-down"
              role="img"
            >
              <svg
                aria-hidden="true"
                class=""
                data-icon="down"
                fill="currentColor"
                focusable="false"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
                />
              </svg>
            </span>
          </div>
          <div
            data-testid="isShow1"
            style="display: none;"
          >
            <div>
              <div>
                <label
                  class="ant-checkbox-wrapper"
                >
                  <span
                    class="ant-checkbox"
                  >
                    <input
                      class="ant-checkbox-input"
                      data-testid="checkbox0"
                      type="checkbox"
                      value=""
                    />
                    <span
                      class="ant-checkbox-inner"
                    />
                  </span>
                  <span>
                    超声
                  </span>
                </label>
              </div>
              <div>
                <label
                  class="ant-checkbox-wrapper"
                >
                  <span
                    class="ant-checkbox"
                  >
                    <input
                      class="ant-checkbox-input"
                      data-testid="checkbox1"
                      type="checkbox"
                      value=""
                    />
                    <span
                      class="ant-checkbox-inner"
                    />
                  </span>
                  <span>
                    心电图
                  </span>
                </label>
              </div>
              <div>
                <button
                  class="ant-btn"
                  type="button"
                >
                  <span>
                    取 消
                  </span>
                </button>
                <button
                  class="ant-btn ant-btn-primary"
                  type="button"
                >
                  <span>
                     确定 
                  </span>
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
